File: src/codes/forced-compatibility/py/ksrc.py
# レガシーAPI（旧仕様）
class LegacyFileSaver:
    def save_data(self, data: str):
        print(f"保存形式: <<<{data}>>>")

class ExportService:
    def export_report(self, json_obj: dict):
        legacy_saver = LegacyFileSaver()
        formatted = str(json_obj).replace("{", "[").replace("}", "]")
        legacy_saver.save_data(formatted)

File: src/codes/forced-compatibility/php/ksrc.php
<?php

// レガシーAPI（旧仕様）
class LegacyFileSaver {
  public function saveData(string $data): void {
    echo "保存形式: <<<{$data}>>>\n";
  }
}

class ExportService {
  public function exportReport(array $json): void {
    $legacySaver = new LegacyFileSaver();
    $formatted = str_replace(['{', '}'], ['[', ']'], json_encode($json));
    $legacySaver->saveData($formatted);
  }
}
